src/states/node.rs

```diff
@@ -2169,6 +2162,10 @@ impl Node {
             .get_peer(pub_id)
             .map_or(false, Peer::is_routing)
     }
+
+    pub fn has_resource_proof_candidate(&self) -> bool {
```

Comment:
I'd have matched the inner method name here to just `has_candidate` and if needed explicit difference eventually then name them throught accordingly.

Reply:
Will rename has_candidate.